suitcase/structure.py

```diff
@@ -85,6 +86,11 @@ def unpack_stream(self, stream):
         logic present for dealing with checksum fields.
 
         """
+        def is_substructure(field):
+            return isinstance(field, SubstructureField) or \
+                    isinstance(field, ConditionalField) and \
+                    isinstance(field.field, SubstructureField)

```

Comment:
Yeah, this check is kind of a hack and I only really tested for the Substructure-in-Conditional case because that was the first obvious problem. If you've got a way to do this generally, by all means.